### DISCUSSION TITLE
Versions & dependencies management

### DISCUSSION THREAD

**Discussion by sunnychaganty** [nandorojo/solito#114] [Q&A]
Team: amazing job on solito!

We are starting to explore this as mono-repo platform for a large scale mobile + web app. I have few questions on how to manager the versions & dependencies across the various folders:

- How do we keep the package.json up to date with the various versions + dependencies? For example: react-native versions across /apps/* & /packages/app etc.?
- Expo has a expo doctor --fix-dependencies command which helps in resolving the correct structure. Would it be possible to have something similar to help ensure the mono-repo structure is a little less crazy? 
- Would you recommend npm shrinkwrap across a this mono-repo? 

Thanks!

**Comment by nandorojo** [comment]
i have some thoughts, i’ll reply tomorrow.

for the first point — see the readme of the starter monorepo!

**Reply by sunnychaganty** [reply]
Thank you, appreciate all the help!

Yes, this is a problem especially for large codebases - while there is a huge benefit of mono-repo, the dependencies are the usual problem & if there is some CLI which can help fix dependencies correctly. 

Looking at the README of the starter mono-repo, while individual package installs are manageable by what is outlined there, its the longer term update of the dependencies which will be problematic. 

For example, I tried to use the lerna update wizard & have this large list of dependencies to be updated for the starter & see this large list, which I am sure developers will mess up in a large team:

![image](https://user-images.githubusercontent.com/1699056/180226879-75cd76b5-429f-4198-9cc6-172ade782cd4.png)

Expo (while a nightmare with dependencies!) has a `expo doctor` utility which helps to fix & ensure the versions will work in the app. So, if there is any such utility in Solito, I think the value would be massive especially for larger codebases to adopt & use for production.

**Reply by nandorojo** [reply]
you can search in the update wizard though so i don’t see the problem with updates. you shouldn’t be updating more than one at a time in my opinion
